PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -3980,21 +3983,21 @@ void ScintillaEditView::hideLines()
 		// Special case: user wants to hide every line in between other hidden sections.
 		// Both "while" loops are executed (merge with above AND below hidden section):
 
-		while (scope == 0)
+		while (scope == 0 && startMarker < nbLines)
 			removeMarker(--startMarker, 1 << MARK_HIDELINESBEGIN);
 
-		while (scope != 0)
+		while (scope != 0 && endMarker < nbLines)

```

Comment:
Even it can stop the infinit loop, it doesn't make sense to me.
Could you explain the situation in the comment:
```cpp
// When hiding lines just below/above other hidden lines,
// merge them into one hidden section:
```
?

Reply:
When line number N is hidden, and the user wants to hide the N-1 or the N+1 line number, then the algorithm should extend the previous hidden section.